sdk/monitor/monitor-ingestion/src/logsIngestionClient.ts

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { TokenCredential } from "@azure/core-auth";
+import { CommonClientOptions } from "@azure/core-client";
+import { GeneratedMonitorIngestionClient } from "./generated";
+import { UploadLogsError, UploadOptions, UploadResult, UploadStatus } from "./models";
+import { GZippingPolicy } from "./gZippingPolicy";
+import { asyncPool } from "./utils/concurrentPoolHelper";
+import { splitDataToChunks } from "./utils/splitDataToChunksHelper";
+
+/**
+ * Options for Montior Logs Ingestion Client
+ */
+export interface LogsIngestionClientOptions extends CommonClientOptions {
+  /** Api Version */
+  apiVersion?: string;
+}
+const defaultIngestionScope = "https://monitor.azure.com//.default";
+
+/**
+ * Client for Monitor Logs Ingestion
+ */
+export class LogsIngestionClient {
+  /**
+   * Overrides client endpoint.
+   */
+  private endpoint: string;
+  private _dataClient: GeneratedMonitorIngestionClient;
+  /**
+   * Construct a MonitorIngestionClient that can be used to query logs using the Log Analytics Query language.
+   *
+   * @param tokenCredential - A token credential.
+   * @param options - Options for the MonitorIngestionClient.
+   */
+  constructor(
+    endpoint: string,
+    tokenCredential: TokenCredential,
+    options?: LogsIngestionClientOptions
+  ) {
+    const credentialOptions = {
+      credentialScopes: defaultIngestionScope,
+    };
+    this.endpoint = endpoint;
+    this._dataClient = new GeneratedMonitorIngestionClient(tokenCredential, this.endpoint, {
+      ...options,
+      credentialScopes: credentialOptions?.credentialScopes ?? defaultIngestionScope
```

Comment:
but with this, the users won't be able to pass in a custom scope right?

Reply:
Today I don't think they can, right? Since you're overriding it with a local object in all cases?

Reply:
Like this `credentialOptions` comes from line 41 and not from the options being passed in

Reply:
Removed reading from credentialOptions right now, since I don't think it's supported. Will check with service team and add reading user passed options if needed. Tracing in the same issue - https://github.com/Azure/azure-sdk-for-js/issues/22615